build_rsas_from_etas.gs

```diff
@@ -0,0 +1,390 @@
+/*
+    Name:        WeDiscover - ETA To RSA Builder, Google Ads Script
+    Description: A script to build RSA ad copy from existing ETA ad
+                 copy.
+    License:     https://github.com/we-discover/public/blob/master/LICENSE
+    Version:     1.0.0
+    Released:    2021-12-13
+    Contact:     scripts@we-discover.com
+*/
+
+// EDIT ME -- Check for RSAs in Paused Ads, Ad Groups or Campaigns
+var checkPausedCampaigns = false;
+var checkPausedAdGroups = false;
+var includePausedRsas = false;
+var pullFromPausedEtas = false;
+
+// Script entrypoint
+function main(){
+  var executionContext = getExecutionContext();
+  var topLevelAccountName = AdsApp.currentAccount().getName();
+  var topLevelAccountId = AdsApp.currentAccount().getCustomerId();
+  var ssName = topLevelAccountName + " (" + topLevelAccountId + ") | ETA to RSA Builder";
+  
+  Logger.log("Making Google Sheet...");
+  var ss = SpreadsheetApp.create(ssName);
+  
+  // If MCC, run data process on a loop through all accounts
+  if (executionContext === 'manager_account') {
+    var accountIterator = AdsManagerApp.accounts().get();
+    while(accountIterator.hasNext()) {
+      AdsManagerApp.select(accountIterator.next());
+      
+      var accountName = AdsApp.currentAccount().getName();
+      var accountId = AdsApp.currentAccount().getCustomerId();
+      var accountNameAndId = accountName + " (" + accountId + ")";  
+      var successCount = 0;
+      
+      Logger.log("Looking in account " + accountNameAndId + " ...");
+      var assets = getAssets(accountName, accountId);
+      if (assets !== 'no_rsas') {
+        var sheet = ss.insertSheet(accountNameAndId)
+        var maxNoOfAssets = getMaxAssets(assets);
+        writeToSheet(sheet, assets, maxNoOfAssets, accountName, accountId);
+        
+        successCount++;
+      }
+      
+      else if (assets === 'no_rsas' && accountIterator.hasNext()) {
+        Logger.log("Trying next account...");
+      }
+      else if (assets === 'no_rsas' && !accountIterator.hasNext()) {
+        Logger.log("No more accounts found. Terminating script.");
+      }   
+       
+      Logger.log("--------------------------------------------------");
+    }
+    
+    // Remove first sheet, which is blank
+    ss.deleteSheet(ss.getSheets()[0]);
+    if (successCount === 0) {
+      Logger.log("No ad groups missing RSAs found. Google Sheet can be safely deleted.");
+    }
+    Logger.log("Script run complete. Google Sheet location: " + ss.getUrl());
+  }
+  
+  // If child account, process on that account only
+  else if (executionContext === 'client_account') {
+    var accountName = AdsApp.currentAccount().getName();
+    var accountId = AdsApp.currentAccount().getCustomerId();
+    var accountNameAndId = accountName + " (" + accountId + ")";
+    var sheet = ss.getSheets()[0].setName(accountNameAndId);
```

Comment:
Why don't you create the sheet only when you find RSAS? or delete the sheet automatically if no rsas are found? Could this potentially create hundreds of empty gsheets if nobody manually deletes them?

Reply:
Yeah, this is a good point. I can't delete the sheet without including G Drive scope, but I will look at only creating the sheet if RSAs are found.

People will only run as a one off though, so even if empty sheets are made it shouldn't be too many

Reply:
Ok. Marking this as resolved and we can add this change in next version. @alexh-wediscover can you add to our asana?